articles/152_roslaunch_multi_machine_launch.md

```diff
@@ -0,0 +1,310 @@
+- Sharing and synchronizing files across machines
+
+## Considerations
+
+There are some outstanding issues that may complicate things:
+
+- How to group nodes/participants/processes is somewhat of an open issue with potential
+   implications for this part of ROS2.
+    - https://github.com/ros2/design/pull/250/files/8ccaac3d60d7a0ded50934ba6416550f8d2af332?short_path=dd776c0#diff-dd776c070ecf252bc4dcc4b86a97c888
+    - The number of domain participants is limited per vendor (Connext is 120 per domain).
+- No `rosmaster` means there is no central mechanism for controlling modes or
+  distributing parameters
+- Machines may be running different operating systems
+- If we intend to do any kind of load balancing, certain types of resources may
+    need to be transferred to other machines.
+     - Calibration data, map files, training data, etc.
+     - Need to keep track of which machine has the most recent version of such
+        resources
+- Security: we'll need to manage credentials across numerous machines both for SSH
+    and secure DDS.
+
+
+## Proposed Multi-Machine Launch Command Line Interface
+
+The multi-machine launching interface is controlled through the `launcher`
+command for the `ros2` command-line tool.  The existing `launch` command
+provides a subset of this functionality that is sufficient for single-machine
+launching.
+
+### Commands
+
+```bash
+$ ros2 launcher
+usage: ros2 launcher [-h] Call `ros2 launcher <command> -h` for more detailed usage. ...
+
+Various launching related sub-commands
+
+optional arguments:
+  -h, --help            show this help message and exit
+
+Commands:
+  launch       Run a launch file
+  list         Search for and list running launch systems
+  attach       Attach to a running launch system and wait for it to finish
+  term         Terminate a running launch system
+
+  Call `ros2 launcher <command> -h` for more detailed usage.
+```
+
+#### `launch`
+
+The `ros2 launcher launch` is equivalent to `ros2 launch`, which is preserved
+for backwards compatibility and ease of use.  It is used to run a launch file.
+
+```bash
+$ ros2 launcher launch -h
+usage: ros2 launcher launch [-h] [-d] [-D] [-p | -s] [-a]
+                            package_name [launch_file_name]
+                            [launch_arguments [launch_arguments ...]] ...
+
+Run a launch file
+
+positional arguments:
+  package_name          Name of the ROS package which contains the launch file
+  launch_file_name      Name of the launch file
+  launch_arguments      Arguments to the launch file; '<name>:=<value>' (for
+                        duplicates, last one wins)
+  argv                  Pass arbitrary arguments to the launch file
+
+optional arguments:
+  -h, --help            Show this help message and exit.
+  -d, --debug           Put the launch system in debug mode, provides more
+                        verbose output.
+  -D, --detach          Detach from the launch process after it has started.
```

Comment:
we would use if launch process kills itself after launch. after the launch just to leave everything else. for embedded platform, we might not need launch process, but we likely to use launch description to init the system.